### PR TITLE
Add downstream tests

### LINKED ISSUE #45: Add downstream tests
As this repo gets used more and more by downstream packages, the risk of accidentally breaking things increases. I think it would be good to add new CI matrix where we run the test suite for ClimaLand, ClimaAtmos (and ClimaCoupler in the future).

### PR DESCRIPTION
**Pull request by Sbozzolo**
Closes #45

### CODE REVIEW

**Comment by Sbozzolo** [comment]
The test that is failing in ClimaLand doesn't have anything to do with ClimaUtilities...

File: .github/workflows/downstream.yml [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+      - uses: julia-actions/cache@v1
+      - uses: julia-actions/julia-buildpkg@v1
+      - uses: actions/checkout@v4
+        with:
+          repository: 'CliMA/${{ matrix.package }}'
+          path: ${{ matrix.package }}
+      - run: |
+          julia --project=${{ matrix.package }} -e 'using Pkg; Pkg.instantiate()'
+          julia --project=${{ matrix.package }} -e 'using Pkg; Pkg.develop(; path = ".")'
+          julia --project=${{ matrix.package }} -e 'using Pkg; Pkg.test()'
```
Comment (juliasloan25) [inline .github/workflows/downstream.yml:38]: can we add `--color=yes` here please?

Comment (Sbozzolo) [reply .github/workflows/downstream.yml]: Added

**Comment by juliasloan25** [comment]
It would be nice to merge this - maybe we can just increase the tolerance for the land test

**Comment by Sbozzolo** [comment]
> It would be nice to merge this - maybe we can just increase the tolerance for the land test

I don't understand how eps(FT) could be 1e-7. 

It is also quite annoying to debug this.

The easiest way out is to increase the tolerance by one order of magntiude again, but this is not very satisfying solution

**Comment by Sbozzolo** [comment]
@juliasloan25 the issue with ClimaLand got fixed (it was `arch = x86` in the action), so I think we can merge this!

**Review by juliasloan25 — APPROVED** [review]

**Pull request merged**